ebs_deploy/commands/deploy_command.py

```diff
@@ -1,17 +1,40 @@
-from ebs_deploy import get, parse_env_config, parse_option_settings, upload_application_archive
+import json
+from datetime import datetime
+
+from ebs_deploy import (get, parse_env_config, parse_option_settings,
+                        upload_application_archive)
+
+TIMEOUT_PER_INSTANCE = 120
 
 
 def add_arguments(parser):
     """
     adds arguments for the deploy command
     """
-    parser.add_argument('-e', '--environment', help='Environment name', required=True)
-    parser.add_argument('-w', '--dont-wait', help='Skip waiting for the init to finish', action='store_true')
-    parser.add_argument('-a', '--archive', help='Archive file', required=False)
-    parser.add_argument('-d', '--directory', help='Directory', required=False)
-    parser.add_argument('-l', '--version-label', help='Version label', required=False)
-    parser.add_argument('-f', '--log-events-to-file', help='Log events to file',
-                        required=False, action='store_true')
+    parser.add_argument('-e', '--environment',
+                        help='Environment name',
+                        required=True)
+    parser.add_argument('-w', '--dont-wait',
+                        help='Skip waiting for the init to finish',
+                        action='store_true')
+    parser.add_argument('-a', '--archive',
+                        help='Archive file',
+                        required=False)
+    parser.add_argument('-d', '--directory',
+                        help='Directory',
+                        required=False)
+    parser.add_argument('-l', '--version-label',
+                        help='Version label',
+                        required=False)
+    parser.add_argument('-f', '--log-events-to-file',
+                        help='Log events to file',
+                        required=False,
+                        action='store_true')
+
+
+def get_env_instances(ec2_conn, env_name):
+    return [r.instances[0] for r in ec2_conn.instances.get_all_reservations()
```

Comment:
This doesn't seem right.
```
ec2_conn = ec2.connect_to_region('us-east-1', aws_access_key_id='AKIAJWRE52Q55RQ3TB5Q', aws_secret_access_key='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')
ec2_srv.instances
```
EC2Connection' object has no attribute 'instances'EC2Connection' object has no attribute 'instances'

Also, you should use AWS filtering and get only instances not reservations, so what you should instead do:
```
return ec2_conn.get_only_instances(filters={'tag:elasticbeanstalk:environment-name': env_name})
```